superset/__init__.py

```diff
@@ -196,40 +196,41 @@ def index(self):
          not FAB's security manager.
          See [4565] in UPDATING.md""")
 
-appbuilder = AppBuilder(
-    app,
-    db.session,
-    base_template='superset/base.html',
-    indexview=MyIndexView,
-    security_manager_class=custom_sm,
-    update_perms=get_update_perms_flag(),
-)
+with app.app_context():
```

Comment:
Tested this - all you need is for the AppBuilder ctor to live inside the app context. The rest can be left outside.

Reply:
We had the same issue if the ConnectorRegistry stuff wasn't included because it starts a bunch of sa.listen() on sqlatables. Are you sure ?  I'll re-test on my end as well.

Reply:
I'm starting the upgrade from superset 0.25.6. The init.py is somewhat different. I wonder if that's the issue.

Reply:
Yea, you're right. On the current master, only the AppBuilder ctor needs to be in the context.

Reply:
This looks really hackish, app_context is there just to enforce the session.close?